src/tools/blast/components/results/HSPDetailPanel.tsx

```diff
@@ -0,0 +1,383 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable @typescript-eslint/camelcase */
+import React, { FC, useCallback, useState } from 'react';
+import ProtvistaTrack from 'protvista-track';
+import ProtvistaNavigation from 'protvista-navigation';
+import ProtvistaMSA from 'protvista-msa';
+import ProtvistaManager from 'protvista-manager';
+import { Loader, DropdownButton, CloseIcon } from 'franklin-sites';
+import { uniq } from 'lodash-es';
+import SlidingPanel from '../../../../shared/components/layouts/SlidingPanel';
+import { BlastHsp } from '../../types/blastResults';
+import { loadWebComponent } from '../../../../shared/utils/utils';
+import useDataApi from '../../../../shared/hooks/useDataApi';
+import { UniProtkbAPIModel } from '../../../../uniprotkb/adapters/uniProtkbConverter';
+import apiUrls from '../../../../uniprotkb/config/apiUrls';
+import FeatureType from '../../../../uniprotkb/types/featureType';
+import { processFeaturesData } from '../../../../uniprotkb/components/protein-data-views/FeaturesView';
+import ErrorHandler from '../../../../shared/components/error-pages/ErrorHandler';
+import {
+  MsaColorScheme,
+  msaColorSchemeToString,
+} from '../../../config/msaColorSchemes';
+
+loadWebComponent('protvista-navigation', ProtvistaNavigation);
+loadWebComponent('protvista-track', ProtvistaTrack);
+loadWebComponent('protvista-msa', ProtvistaMSA);
+loadWebComponent('protvista-manager', ProtvistaManager);
+
+type UniProtkbAccessionsAPI = {
+  results: UniProtkbAPIModel[];
+};
+
+export type HSPDetailPanelProps = {
+  hsp: BlastHsp;
+  hitAccession: string;
+  onClose: () => void;
+  hitLength: number;
+  queryLength: number;
+};
+
+const HSPDetailPanel: FC<HSPDetailPanelProps> = ({
+  hsp,
+  hitAccession,
+  onClose,
+  hitLength,
+  queryLength,
+}) => {
+  const {
+    hsp_align_len,
+    hsp_query_from,
+    hsp_query_to,
+    hsp_qseq,
+    hsp_hit_from,
+    hsp_hit_to,
+    hsp_hseq,
+    hsp_identity,
+  } = hsp;
+  // TODO calculate actual length based on total match and query lengths
+  // const actualLength = hsp_align_len;
+  const actualLength = hitLength;
```

Comment:
As hitLength is a prop, this is now superfluous, just use `hitLength`

Reply:
Done